ops/terraform/services/server/server-regression/main.tf

```diff
@@ -64,3 +74,66 @@ resource "aws_sqs_queue" "this" {
   visibility_timeout_seconds = local.lambda_timeout_seconds
   kms_master_key_id          = local.kms_key_id
 }
+
+resource "aws_sqs_queue" "pipeline_signal" {
+  name                       = local.pipeline_signal_queue_name
+  visibility_timeout_seconds = local.lambda_timeout_seconds
+  kms_master_key_id          = local.kms_key_id
+}
+
+resource "aws_glue_crawler" "this" {
+  name = "${local.insights_database}-${local.service}"
+  tags = merge(local.shared_tags, { Name = "${local.insights_database}-${local.service}", application = "bfd-insights" })
+
+  database_name = local.insights_database
+
+  role = data.aws_iam_role.insights.arn
+
+  lineage_configuration {
+    crawler_lineage_settings = "DISABLE"
+  }
+
+  recrawl_policy {
+    recrawl_behavior = "CRAWL_NEW_FOLDERS_ONLY"
+  }
+
+  s3_target {
+    path = "s3://${data.aws_s3_bucket.insights.id}/databases/${local.insights_database}/${local.insights_table}"
+  }
+
+  schema_change_policy {
+    delete_behavior = "LOG"
+    update_behavior = "LOG"
+  }
+}
+
+resource "aws_lambda_permission" "allow_s3_glue_trigger" {
+  statement_id  = "bfd-${local.env}-${local.service}-glue-trigger-allow-s3"
+  action        = "lambda:InvokeFunction"
+  function_name = aws_lambda_function.glue_trigger.arn
+  principal     = "s3.amazonaws.com"
+  source_arn    = data.aws_s3_bucket.insights.arn
+}
+
+resource "aws_lambda_function" "glue_trigger" {
+  description   = "Triggers the bfd-${local.env}-${local.service} Glue Crawler to run when new statistics are uploaded to S3"
+  function_name = "bfd-${local.env}-${local.service}-glue-trigger"
+  tags          = merge(local.shared_tags, { Name = "bfd-${local.env}-${local.service}-glue-trigger" })
+  kms_key_arn   = local.kms_key_arn
+
+  filename         = data.archive_file.glue_trigger.output_path
+  source_code_hash = data.archive_file.glue_trigger.output_base64sha256
+  architectures    = ["x86_64"]
+  handler          = "glue-trigger.handler"
+  memory_size      = 128
+  package_type     = "Zip"
+  runtime          = "python3.9"
+  timeout          = local.glue_trigger_lambda_timeout_seconds
+  environment {
+    variables = {
+      CRAWLER_NAME = aws_glue_crawler.this.name
+    }
+  }
+
+  role = aws_iam_role.glue_trigger.arn
+}
```

Comment:
Yep, the crawler will auto-generate the table based on the name of the parent "directory" of the Glue Crawler's path. This behavior is being implicitly relied upon by the stats loading process, and I agree that it should probably be explicit so we don't run into issues down the line.

I'll add an explicit `resource` for the Glue Catalog Tables generated by the Glue Crawlers.

Reply:
For reasons specified in my comment above, I think we should table this conversation for later. After exploring this for a bit I do not think there is any benefit beyond making the connection between the Glue Crawler and Glue Catalog Table explicit. I actually think this might be detrimental, as it is only possible to specify the "Crawl new folders only" recrawl behavior with S3 targets.

Reply:
You can achieve the same result, I think, by enabling [bookmarks](https://docs.aws.amazon.com/glue/latest/dg/monitor-continuations.html) on a Glue job, but I'm not sure there's anything similar for a Lambda. There probably isn't. Thanks for looking into it. Sorry if it was an unnecessary tangent!